### ISSUE TITLE
Reproducing 3D segmentation with 2D stacks

### ISSUE THREAD

**Issue by jackyko1991** [MouseLand/cellpose#637]
Hello all,

Is there any python interface provides 3D stacking to replicate the Figure 5 in original Cellpose publication? 

![image](https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41592-020-01018-x/MediaObjects/41592_2020_1018_Fig6_HTML.png?as=webp)
a. A 3D volume is sliced into xy, xz and yz sections. 
b. Example frames for each type of section. 
c. Each 2D frame is passed through the Cellpose prediction network (trained on generalized data, Fig. [4f,g](https://www.nature.com/articles/s41592-020-01018-x#Fig4)), generating predicted 2D gradient maps. 
d. The six predicted gradient maps (two each for x, y and z) are pairwise averaged and combined into a single 3D gradient vector field xyz. Shown are the combined xy gradient map as well as the z gradient map, for a single xy slice through the volume. 
e. The 3D gradient maps are used to create pixel dynamics toward the 3D sinks of the gradient vector field, and the pixels are grouped into 3D masks if they converge to the same sink. 
f. Example xy slices with ground-truth (blue) and predicted (dotted yellow) masks at different depths in the z stack. 
g. Average precision at different IoU thresholds of Cellpose3D compared to other models (n = 183 cells in test volume).

**Comment by carsen-stringer** [comment]
To run in the GUI the 3D approach in a-e, drag and drop your stack in the GUI and segment it by pressing on one of the model buttons. Make sure the `stitch_threshold` is off. 

From the CLI, run `python -m cellpose --dir path/to/files --do_3D`.

To run in a jupyter notebook, check out the last section of this notebook, using `model.eval(img_3D, do_3D=True)` : https://github.com/MouseLand/cellpose/blob/main/notebooks/run_cellpose_GPU.ipynb. The notebook will make the plot in f for you automatically.
